### PR TITLE
Preformatted text

### PR DESCRIPTION
**Pull request by ehrencrona**
Maintain whitespace and newlines as-in inside `<pre>` tags, i.e. formatting

```
<pre>
  _______
 /  12   \
|    |    |
|9   |   3|
|     \   |
|         |
 \___6___/
</pre>
```

does not turn into 

```
<pre> _______ / 12 \\ | | | |9 | 3| | \\ | | | \\___6___/</pre>
```

Also consider attribute values (with one exception: the `class` attribute) to be pre-formatted. Do not add or remove line breaks there.

This builds on PR #120. I thought it sensible to nevertheless keep it as a separate PR.

I got some code from #41, e.g. the `isPreTagContent` function.

Relates to #28 . I chose not to pursue the idea of configuring which tags are white-space sensitive. I'm happy to add that if there is a feeling it should be added; in any case it can easily be added on top of this PR.

### CODE REVIEW

**Comment by dummdidumm** [comment]
I'm a little confused now on what to review first 😄 does this PR contain everything of #120 + the additional `<pre>` behavior?

**Comment by ehrencrona** [comment]
@dummdidumm Yes it does. It probably makes sense to first review #120. If that is ok, this one (#127) can be reviewed and #116 can be closed. 

If #120 is rejected, #127 can be closed (because it builds on it) but it's still worth taking a look at #116.

File: src/print/index.ts [hunk 1]
```diff
@@ -478,6 +494,8 @@ function printChildren(path: FastPath, print: PrintFn): Doc[] {
     // the index of the last child doc we could add a linebreak after
     let lastBreakIndex = -1;
 
+    let isPreformat = isPreTagContent(path);
```
Comment (dummdidumm) [inline src/print/index.ts:497]:
```suggestion
    const isPreformat = isPreTagContent(path);
```

**Review by dummdidumm — APPROVED** [review]
One minor suggestion, that's all. 👍

**Pull request merged**